cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -749,6 +766,14 @@ public void startSchedule(long basePollSize, long basePollTs, long lastRunSize,
       this.lastRunSize = lastRunSize;
       this.lastRunTs = lastRunTs;
       this.active.set(active);
+      if (persist) {
+        scheduleStore.persist(programId, programType, streamSizeSchedule, basePollSize, basePollTs,
```

Comment:
There are two cases here, yes: either I just read the schedule info from the store, and then ``persist`` will be false and I won't store anything, or we create a new schedule. In that last case, basePollSize and Ts will be initialized with the values obtained from a polling - see ``createScheduleTask()``. This schedule needs to be persisted right at its creation, hence doing it here.